src/components/networked-hand-controls.js

```diff
@@ -153,38 +113,284 @@ AFRAME.registerComponent('networked-hand-controls', {
     if (oldData.visible != this.data.visible) {
       this.el.object3D.visible = this.data.visible;     
     }
+
+    // this block is for handling updates() from hand model to controller model, or vice versa.
+    if (oldData.handModelStyle &&
+        oldData.handModelStyle !== this.data.handModelStyle
+      ) {
+      // first, remove old model
+      this.el.removeObject3D(this.str.mesh);
+      ['gltf-model','obj-model'].forEach(modelComponent => {
+        if (this.el.components[modelComponent]) {
+          this.el.removeAttribute(modelComponent);
+        }
+      })
+      // then, add correct model
+      if (this.data.handModelStyle !== "controller") {
+        this.addHandModel();
+      }
+      else {
+        // it would be nice if this worked, but those components don't handle this option in their update() methods:
+        // this.el.setAttribute(this.data.controllerComponent, 'model', true)
+
+        // so we first remove the controller component as best we can
+        if (this.el.components[this.data.controllerComponent]) {
+          // there is generally no remove() specified, but they do have pause(), which is close enough:
+          try {
+            this.el.components[this.data.controllerComponent].pause();
+            this.el.components[this.data.controllerComponent].remove();
+          } catch(e) {
+            console.error("NAF: possible problem while updating handModelStyle to controller", e)
+          }
```

Comment:
This try catch can be removed completely right? removeAttribute below is already doing the job.